### PR TITLE
Add network info

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
related https://github.com/EveripediaNetwork/issues/issues/108

### CODE REVIEW

**Comment by Emmanueldmlr** [comment]
@JustinCast  @kesar , kindly review.

**Comment by kesar** [comment]
image


there is a weird amount of txs happening here. something must be wrongly memozide

**Review by kesar — APPROVED** [review]

**Comment by JustinCast** [comment]
@Emmanueldmlr I'm connected to Rinkeby, but the dialog keeps saying "Ethereum Network" 
image

File: src/components/Layout/Network/NetworkErrorNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const NetworkErrorNotification = ({
+  onClose,
+  isOpen,
+  switchNetwork,
+}: {
+  isOpen: boolean
+  onClose: () => void
+  switchNetwork: () => void
+}) => {
+  const cancelRef = React.useRef<FocusableElement>(null)
```
Comment (JustinCast) [inline src/components/Layout/Network/NetworkErrorNotification.tsx:24]: we can import useRef instead of doing React.useRef

Comment (Emmanueldmlr) [reply src/components/Layout/Network/NetworkErrorNotification.tsx:24]: Thanks for pointing this out. I will fix it.

File: src/components/Layout/Network/NetworkMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import detectEthereumProvider from '@metamask/detect-provider'
+import { Network, Networks } from '@/data/NetworkData'
+import networkMap from '@/utils/networkMap'
+import NetworkErrorNotification from '@/components/Layout/Network/NetworkErrorNotification'
+
+const NetworkMenu = () => {
+  const [currentNetwork, setCurrentNetwork] = useState<Network>(Networks[0])
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [detectedProvider, setDetectedProvider] = useState<any>()
```
Comment (JustinCast) [inline src/components/Layout/Network/NetworkMenu.tsx:23]: We can avoid setting "any" types

File: src/components/Layout/Network/NetworkMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    (chainDetails: string) => {
+      if (chainDetails !== chainId) {
+        onOpen()
+      }
+    },
+    [onOpen, chainId],
+  )
+
+  useEffect(() => {
+    const getConnectedChain = async (provider: any) => {
```
Comment (JustinCast) [inline src/components/Layout/Network/NetworkMenu.tsx:36]: Same here

File: src/components/Layout/Network/NetworkMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      const connectedChainId = await provider.request({
+        method: 'eth_chainId',
+      })
+      handleChainChanged(connectedChainId)
+    }
+
+    const getDetectedProvider = async () => {
+      const provider: any = await detectEthereumProvider()
```
Comment (JustinCast) [inline src/components/Layout/Network/NetworkMenu.tsx:44]: Same here

File: src/components/Layout/Network/NetworkMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  }
+
+  const handleSwitchNetwork = async () => {
+    try {
+      await detectedProvider.request({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId }],
+      })
+      onClose()
+    } catch (switchError: any) {
```
Comment (JustinCast) [inline src/components/Layout/Network/NetworkMenu.tsx:76]: Avoid "any" type

File: src/components/Layout/Network/NetworkMenu.tsx (hunk 2 above)
Comment (Emmanueldmlr) [reply src/components/Layout/Network/NetworkMenu.tsx:23]: I can't seem to get the type from the provider

Comment (kesar) [reply src/components/Layout/Network/NetworkMenu.tsx:23]: Yeah, it must be hard. I guess one solution its to create a type that has all the method we need from the provider for all the actions where its called, then we can use that 👍🏻

Comment (Emmanueldmlr) [reply src/components/Layout/Network/NetworkMenu.tsx:23]: Okay then. I will do that. Thanks.

**Pull request merged**